File: zillow-home-value/zillow-home-value/zillow_data_manage.py
import numpy as np
import pandas as pd
import os

os.chdir('C:\\Users\\vinay.benny\\Documents\\Kaggle\\kaggling-repo\\zillow-home-value\\zillow-home-value')

# Read the datasets
properties = pd.read_csv("../data/properties_2016.csv")
transactions = pd.read_csv("../data/train_2016.csv", parse_dates = ["transactiondate"] )
submission = pd.read_csv("../data/sample_submission.csv")

print(transactions.shape)
print(properties.shape)

# Rename properties and group columns to be more meaningful
properties.rename(columns = {
    'parcelid':'id_parcel',
    'yearbuilt':'build_year',
    'basementsqft':'area_basement',
    'yardbuildingsqft17':'area_patio',
    'yardbuildingsqft26':'area_shed',
    'poolsizesum':'area_pool',
    'lotsizesquarefeet':'area_lot',
    'garagetotalsqft':'area_garage',
    'finishedfloor1squarefeet':'area_firstfloor_finished',
    'calculatedfinishedsquarefeet':'area_total_calc',
    'finishedsquarefeet6':'area_base',
    'finishedsquarefeet12':'area_live_finished',
    'finishedsquarefeet13':'area_liveperi_finished',
    'finishedsquarefeet15':'area_total_finished',
    'finishedsquarefeet50':'area_unknown',
    'unitcnt':'num_unit',
    'numberofstories':'num_story',
    'roomcnt':'num_room',
    'bathroomcnt':'num_bathroom',
    'bedroomcnt':'num_bedroom',
    'calculatedbathnbr':'num_bathroom_calc',
    'fullbathcnt':'num_bath',
    'threequarterbathnbr':'num_75_bath',
    'fireplacecnt':'num_fireplace',
    'poolcnt':'num_pool',
    'garagecarcnt':'num_garage',
    'regionidcounty':'region_county',
    'regionidcity':'region_city',
    'regionidzip':'region_zip',
    'regionidneighborhood':'region_neighbor',
    'taxvaluedollarcnt':'tax_total',
    'structuretaxvaluedollarcnt':'tax_building',
    'landtaxvaluedollarcnt':'tax_land',
    'taxamount':'tax_property',
    'assessmentyear':'tax_year',
    'taxdelinquencyflag':'tax_delinquency',
    'taxdelinquencyyear':'tax_delinquency_year',
    'propertyzoningdesc':'zoning_property',
    'propertylandusetypeid':'zoning_landuse',
    'propertycountylandusecode':'zoning_landuse_county',
    'fireplaceflag':'flag_fireplace',
    'hashottuborspa':'flag_tub',
    'buildingqualitytypeid':'quality',
    'buildingclasstypeid':'framing',
    'typeconstructiontypeid':'material',
    'decktypeid':'deck',
    'storytypeid':'story',
    'heatingorsystemtypeid':'heating',
    'airconditioningtypeid':'aircon',
    'architectural_style':'architecturalstyletypeid',
    'pooltypeid10':'spa',
    'pooltypeid2':'pool_with_spa',
    'pooltypeid7':'pool_without_hottub'
    }, inplace = True)


transactions.rename(columns = {'parcelid' : 'id_parcel', 'transactiondate' : 'date'}, inplace = True)
submission.rename(columns = {'ParcelId' : 'id_parcel'}, inplace = True)

# Transform variables
properties['tax_delinquency'] = np.where( properties['tax_delinquency'] == "Y", 1, 0 )
properties['flag_fireplace'] = np.where( properties['flag_fireplace'] == "true", 1, 0 )
properties['flag_tub'] = np.where( properties['flag_tub'] == "true", 1, 0 )
properties['tax_delinquency_year'] = np.where( properties['tax_delinquency_year'] > 20, 
          1900 + properties['tax_delinquency_year'],  2000 + properties['tax_delinquency_year'] )
properties['tax_delinquency_year'] = np.where( pd.isnull(properties['tax_delinquency_year']), 2016, properties['tax_delinquency_year'])

transactions['month'] = transactions['date'].dt.month
transactions.drop('date', inplace = True, axis = 1)


# Recode some column datatypes
properties[['region_city', 'region_county', 'region_neighbor', 'region_zip', 'zoning_landuse_county', 
'zoning_landuse', 'zoning_property', 'quality', 'framing', 'material', 'deck', 'story', 'heating','aircon', 
'architecturalstyletypeid','spa', 'pool_with_spa', 'pool_without_hottub', 'fips']] = properties[['region_city', 
'region_county', 'region_neighbor', 'region_zip', 'zoning_landuse_county', 'zoning_landuse', 'zoning_property', 
'quality', 'framing', 'material', 'deck', 'story', 'heating','aircon', 'architecturalstyletypeid','spa', 
'pool_with_spa', 'pool_without_hottub', 'fips']].apply(pd.Categorical)



# Join the datasets together
transactions = pd.merge(transactions, properties, on = "id_parcel", how="inner")

print(transactions.shape)
transactions.head(2).transpose()
transactions.dtypes

File: zillow-home-value/zillow-home-value-R/zillow-data-manage.R
  library(tidyr)
  library(data.table)
  library(dplyr)
  library(ggplot2)
  library(lubridate)
  
  set.seed(12345)
  
  # Set working directory
  setwd("C:/Users/vinay.benny/Documents/Kaggle/kaggling-repo/zillow-home-value/zillow-home-value-R")
  
  
  ############################### Data Extraction ################################################
  
  # Read files into memory
  properties <- fread('../data/properties_2016.csv', stringsAsFactors = FALSE)
  transactions <- fread('../data/train_2016.csv', stringsAsFactors = FALSE)
  
  # Rename columns to more meaningful values
  properties <- properties %>% rename(
    id_parcel = parcelid,
    build_year = yearbuilt,
    area_basement = basementsqft,
    area_patio = yardbuildingsqft17,
    area_shed = yardbuildingsqft26, 
    area_pool = poolsizesum,  
    area_lot = lotsizesquarefeet, 
    area_garage = garagetotalsqft,
    area_firstfloor_finished = finishedfloor1squarefeet,
    area_total_calc = calculatedfinishedsquarefeet,
    area_base = finishedsquarefeet6,
    area_live_finished = finishedsquarefeet12,
    area_liveperi_finished = finishedsquarefeet13,
    area_total_finished = finishedsquarefeet15,  
    area_unknown = finishedsquarefeet50,
    num_unit = unitcnt, 
    num_story = numberofstories,  
    num_room = roomcnt,
    num_bathroom = bathroomcnt,
    num_bedroom = bedroomcnt,
    num_bathroom_calc = calculatedbathnbr,
    num_bath = fullbathcnt,  
    num_75_bath = threequarterbathnbr, 
    num_fireplace = fireplacecnt,
    num_pool = poolcnt,  
    num_garage = garagecarcnt,  
    region_county = regionidcounty,
    region_city = regionidcity,
    region_zip = regionidzip,
    region_neighbor = regionidneighborhood,  
    tax_total = taxvaluedollarcnt,
    tax_building = structuretaxvaluedollarcnt,
    tax_land = landtaxvaluedollarcnt,
    tax_property = taxamount,
    tax_year = assessmentyear,
    tax_delinquency = taxdelinquencyflag,
    tax_delinquency_year = taxdelinquencyyear,
    zoning_property = propertyzoningdesc,
    zoning_landuse = propertylandusetypeid,
    zoning_landuse_county = propertycountylandusecode,
    flag_fireplace = fireplaceflag, 
    flag_tub = hashottuborspa,
    quality = buildingqualitytypeid,
    framing = buildingclasstypeid,
    material = typeconstructiontypeid,
    flag_deck = decktypeid,
    story = storytypeid,
    heating = heatingorsystemtypeid,
    aircon = airconditioningtypeid,
    architectural_style= architecturalstyletypeid,
    flag_spa = pooltypeid10,
    flag_pool_with_spa = pooltypeid2,
    flag_pool_without_hottub = pooltypeid7
  ) 
  
  # Apply datatypes to variables
  properties <- properties %>%
      mutate(
          region_county = as.factor(region_county),
          region_city = as.factor(region_city),
          region_zip = as.factor(region_zip),
          region_neighbor = as.factor(region_neighbor),
          tax_delinquency = as.factor(tax_delinquency),
          zoning_property = as.factor(zoning_property),
          zoning_landuse = as.factor(zoning_landuse),
          zoning_landuse_county = as.factor(zoning_landuse_county),
          flag_fireplace = as.factor(flag_fireplace), # Binary
          flag_tub = as.factor(flag_tub), # Binary
          quality = as.integer(quality), # Ordinal
          framing = as.factor(framing),
          material = as.factor(material),
          flag_deck = as.integer(flag_deck),# Binary
          flag_story = as.integer(story), # Could be binary, representing presence of basement or not
          heating = as.factor(heating),
          aircon = as.factor(aircon),
          architectural_style = as.factor(architectural_style),
          flag_spa = as.integer(flag_spa), # Binary
          flag_pool_with_spa = as.factor(flag_pool_with_spa), # Binary
          flag_pool_without_hottub = as.factor(flag_pool_without_hottub), # Binary
          fips = as.factor(fips)
          )  
  
  # Add a month variable and rename columns
  transactions <- transactions %>% 
    rename(
      id_parcel = parcelid,
      date = transactiondate) %>% 
    mutate(  month = month(date),
             day = day(date)
    ) 
  
  
  #  Convert the datasets into data.table and let id_parcel as key
  # properties <- data.table(properties)
  # transactions  <- data.table(transactions)
  # setkey(properties, id_parcel)
  # setkey(transactions, id_parcel)
  
  
  # Set column nam variables
  idcol <- "id_parcel"
  intcols <- names(properties[, sapply(properties, is.integer) & !(names(properties) == idcol)])
  catcols <- names(properties[, sapply(properties, is.factor) & !(names(properties) == idcol)])
  numcols <- names(properties[, !names(properties) %in% c(catcols, intcols, idcol) ])
  
  
